torch_xla/csrc/aten_xla_bridge.cpp

```diff
@@ -125,6 +134,28 @@ std::vector<at::Tensor> XlaCreateTensorList(const at::TensorList& tensors) {
   return aten_xla_tensors;
 }
 
+std::vector<c10::optional<at::Tensor>> XlaCreateOptTensorList(
+    const std::vector<c10::optional<at::Tensor>>& tensors) {
+  std::vector<c10::optional<at::Tensor>> opt_aten_xla_tensors(tensors.size());
+  std::vector<at::Tensor> materialized_tensors;
+  std::vector<bool> to_translate(tensors.size());
+  for (size_t i = 0; i < tensors.size(); ++i) {
+    auto tensor = tensors[i];
+    if (tensor.has_value()) {
+      to_translate[i] = true;
+      materialized_tensors.push_back(*tensor);
+    }
+  }
+  auto aten_materialzied_tensors = XlaCreateTensorList(materialized_tensors);
+  for (size_t i = 0, defined_pos = 0; i < tensors.size(); ++i) {
+    opt_aten_xla_tensors[i] =
+        to_translate[i] ? c10::make_optional(std::move(
```

Comment:
I think this gets a bit too crowded, especially because the subtle `defined_pos` increment in one side of the ternary.
I believe the code as in the function above reads a bit better in this case.